feat: Enable Rizzcharts sample and enhance client message handling

doitundoit opened the pull request:
### Summary

This PR enables the rizzcharts sample agent with improved map support and significantly hardens the Lit client's message handling capabilities. It addresses issues with MIME type definitions and improves the client's ability to handle various data formats.


### Changes

#### Rizzcharts Agent

Map Support: Updated `agent.py` and `map.json` to support map images, allowing for either Google Maps API integration or fallback placeholders.

#### Lit Client Improvements

Robust Message Handling:
- Implemented a fallback mechanism to retrieve agent messages from history when needed.
- Added support for parsing string-encoded JSON data parts.

Configuration:
- Fixed the A2UI MIME type constant to `application/json+a2ui`.
- Added debug logging (`console.debug`) for better traceability.

Demo Setup: 
- Added rizzcharts demo scripts and configuration to `samples/client/lit/package.json`.


### Verification
-  Python: Verified syntax and build configuration for rizzcharts agent.
- TypeScript: Successfully built the lit shell client (npm run build:tsc).

sugoi-yuzuru commented on samples/client/lit/shell/client.ts:
```diff
@@ -18,7 +18,7 @@ import { Part, SendMessageSuccessResponse, Task } from "@a2a-js/sdk";
 import { v0_8 } from "@a2ui/lit";
 
-const A2AUI_MIME_TYPE = "application/json+a2aui";
+const A2AUI_MIME_TYPE = "application/json+a2ui";
```
Huge thanks for the fix. 
Could we rename the const to `A2UI_MIME_TYPE` as well? 

Also, it would be great if we can decouple this fix into its own PR to limit the scope of the commit for posterity.

doitundoit commented:
> All changes look good.
> 
> Thank you for noticing the json+a2 aui MIME_TYPE issue. It seems it is more prevalent in the codebase that there are more places needed to be adjusted. Would you kindly isolate this small refactor into its own fix/refactor PR?
> 
> Changes for the lit looks good too. Can we isolate this change into its own PR to keep the scope of the commit as narrow and targetted as possible?
> 
> For Rizzcharts, the change look good too. However, I have some reservations in regards to sourcing image from an external service. Can we simply add a placeholder image within the codebase and use that instead of pointing to a thirdparty service?

Hi @sugoi-yuzuru,

Thank you for the detailed review! I've split the changes from this PR into three smaller, more focused PRs to keep the scope narrow and targeted as requested:

- PR #418: Fixes the A2UI MIME type issues and renames the constant to A2UI_MIME_TYPE for better consistency.
- PR #419: Ported the Lit client robustness improvements, including the history fallback mechanism and string-encoded JSON handling.
- PR #420: Enables Rizzcharts map support. Per your suggestion, I've removed the dependency on external placeholder services (placehold.co) and added a local map-placeholder.png asset to the client's public directory.

I will close this PR now in favor of these individual PRs. Please take a look at them when you have a chance. Thanks again!
